src/librustc_mir/interpret/terminator.rs

```diff
@@ -135,29 +135,29 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     self.goto_block(Some(target))?;
                 } else {
                     // Compute error message
-                    use rustc::mir::interpret::PanicMessage::*;
-                    return match msg {
+                    use rustc::mir::interpret::PanicInfo::*;
+                    match msg {
```

Comment:
This is the only other case (I think) that got more fragile because the `return match` is gone.